tools/codegen/gen.py

```diff
@@ -474,7 +474,18 @@ def go(f: NativeFunction) -> Optional[str]:
 
         dispatcher_returns_type = dispatcher.returns_type(f.func.returns)
         dispatcher_args = dispatcher.arguments(f.func)
-        dispatcher_exprs = dispatcher.legacydispatcherarguments_exprs(legacy_dispatcher_args)
+
+
+        if local.use_c10_dispatcher() is UseC10Dispatcher.full:
+            returns_type = dispatcher_returns_type
+            args = dispatcher_args
+            exprs = dispatcher.legacydispatcherarguments_exprs(args)
```

Comment:
This is a type violation and mypy agrees.

```
AssertionError: mypy failed: tools/codegen/gen.py:482:64: error: Argument 1 to "legacydispatcherarguments_exprs" has incompatible type "Sequence[DispatcherArgument]"; expected "Sequence[LegacyDispatcherArgument]"
tools/codegen/gen.py:486:20: error: Incompatible types in assignment (expression has type "Sequence[LegacyDispatcherArgument]", variable has type "Sequence[DispatcherArgument]")
Found 2 errors in 1 file (checked 1 source file)
```

As the name of this function suggests, this should take legacy dispatcher arguments, and maps them into dispatcher expressions.

In this case, you have dispatcher arguments and you want to map them to dispatcher expressions. This should be quite trivial to do, so just make a new function in dispatcher.py for it.

Reply:
fixed